src/views/Setup.vue

```diff
@@ -1,27 +1,66 @@
 <script setup>
-import { ref, onMounted } from 'vue';
+import { ref, reactive, watch, onMounted, onUnmounted } from 'vue';
 import { useRouter } from 'vue-router';
+import { useEns } from '@/composables/useEns';
+import { useWeb3 } from '@/composables/useWeb3';
+import { useModal } from '@/composables/useModal';
 import { setPageTitle } from '@/helpers/utils';
+import RegisterENS from '@/components/RegisterENS.vue';
 
 const router = useRouter();
-const tlds = ['.eth', '.xyz', '.com', '.org', '.io', '.app', '.art'];
+const { web3, web3Account } = useWeb3();
+const { validEnsTlds, ownedEnsDomains, loadOwnedEnsDomains } = useEns();
+const { modalAccountOpen } = useModal();
 
-const id = ref('');
+onMounted(() => {
+  setPageTitle('page.title.setup');
+});
 
-function isValidTLD(id) {
-  return tlds.some(tlds => id.endsWith(tlds));
-}
+// used either on click on existing owned domain OR once a newly registered
+// domain is returned by the ENS subgraph.
+const goToSettings = key => {
+  router.push({ name: 'spaceSettings', params: { key } });
+};
 
-function handleSubmit() {
-  router.push({
-    name: 'spaceSettings',
-    params: { key: id.value.toLowerCase() }
-  });
-}
+// input for new domain registration
+const newDomain = reactive({
+  name: '',
+  tld: validEnsTlds[0]
+});
 
-onMounted(() => {
-  setPageTitle('page.title.setup');
+// indicates whether to periodically check for new domains
+// (after clicking on "Register")
+const waitingForRegistration = ref(false);
+// handle periodic lookup (every 10s) while registering new domain
+let waitingForRegistrationInterval;
+const waitForRegistration = () => {
+  waitingForRegistration.value = true;
+  clearInterval(waitingForRegistrationInterval);
+  waitingForRegistrationInterval = setInterval(loadOwnedEnsDomains, 10000);
+};
+
+// If after loading domains, we have more than before, there was a new registration.
+// If the new domain matches the one that was input, directly jump to settings page.
+watch(ownedEnsDomains, (newVal, oldVal) => {
+  if (newVal.length > oldVal.length) {
+    waitingForRegistration.value = false;
+    clearInterval(waitingForRegistrationInterval);
+    if (newVal.find(d => d.name === `${newDomain.name}.${newDomain.tld}`)) {
+      goToSettings(`${newDomain.name}.${newDomain.tld}`);
+    }
+  }
+});
+
+// load domains initially and update on account change
+const loadingOwnedEnsDomains = ref(true);
```

Comment:
Can this be moved into the composable? Then line 56 could also be move inside.

Reply:
I was considering the opposite tbh.... removing the composable entirely, since it probably wont be reused anyway.